AppController/djinn.rb

```diff
@@ -4996,6 +4907,52 @@ def find_lowest_free_port(starting_port)
   end
 
 
+  # Updates @app_info_map with registered instances. This must be called under
+  # APPS_LOCK.
+  #
+  # Args:
+  #   version_key: A string specifying the version key to update.
+  def update_registered_instances(version_key)
+    begin
+      zk_instances = ZKInterface.get_children(
+        "/appscale/instances_by_version/#{version_key}")
+    rescue FailedZooKeeperOperationException
+      Djinn.log_warn('Unable to fetch list of registered instances.')
+      return
+    end
+
+    unless zk_instances.empty?
+      @app_info_map[version_key] = {} unless @app_info_map.key?(version_key)
+      unless @app_info_map[version_key].key?('appservers')
+        @app_info_map[version_key]['appservers'] = []
+      end
+    end
+    known_instances = @app_info_map[version_key]['appservers']
```

Comment:
Is there a chance that @app_info_map[version_key]['appservers'] is non-existent? Could zk_instances be empty and also app_info_map not been registered?

Reply:
Good point. There's no reason to avoid populating that if it doesn't exit.